pandas/core/generic.py

```diff
@@ -2129,19 +2189,17 @@ class  max_speed
         numpy.ndarray.take
         numpy.take
         """
+
+    @Appender(_shared_docs['take'])
+    def take(self, indices, axis=0, convert=True, is_copy=True, **kwargs):
         nv.validate_take(tuple(), kwargs)
-        self._consolidate_inplace()
-        new_data = self._data.take(indices,
-                                   axis=self._get_block_manager_axis(axis),
-                                   convert=True, verify=True)
-        result = self._constructor(new_data).__finalize__(self)
 
-        # maybe set copy if we didn't actually change the index
-        if is_copy:
-            if not result._get_axis(axis).equals(self._get_axis(axis)):
-                result._set_is_copy(self)
+        if not convert:
```

Comment:
``convert=None`` should be the default and show the warning if its not None.

Reply:
Why!?  `convert=None` makes less sense compared `convert=False`.

Reply:
the point is to deprecate this. How can you tell that someone is then passing the convert parameter at all? you can then check if ``convert`` is not ``None`` to see if it was passed

Reply:
That's because `convert=True` is the default.